core/src/Kokkos_Macros.hpp

```diff
@@ -428,10 +428,10 @@
 #define KOKKOS_ENABLE_ASM 1
 #endif
 
-#if KOKKOS_COMPILER_GNU < 500
-#define KOKKOS_IMPL_YOLO_ASSUME_TRIVIALLY_COPYABLE_TO_WORK_AROUND_BUG 1
 #endif
 
+#if (__GNUC__ < 5)
```

Comment:
This will be a pain to maintain.  At the very least you would need to add a comment.
I would rather have you keep the old code you removed in `#if defined(KOKKOS_COMPILER_GNU)` and add the new code in `#if defined(KOKKOS_COMPILER_IBM)`.

Reply:
No this can also affect clang, nvcc, PGI and intel. Everything which uses the C++ headers from GNU.

Reply:
Then comment

Reply:
There you go ....